integration_tests/test_new_eo_staclike_metadata_fields.py

```diff
@@ -0,0 +1,241 @@
+                                             'oa_azimuthal_incident', 'oa_combined_terrain_shadow', 'oa_exiting_angle',
+                                             'oa_fmask', 'oa_incident_angle', 'oa_nbar_contiguity',
+                                             'oa_nbart_contiguity', 'oa_relative_azimuth', 'oa_relative_slope',
+                                             'oa_satellite_azimuth', 'oa_satellite_view', 'oa_solar_azimuth',
+                                             'oa_solar_zenith', 'oa_time_delta'),
+                               with_grid_spec=False,
+                               storage=None):
+
+    if storage is None and with_grid_spec is True:
+        storage = {'crs': 'epsg:4326'}
+
+    measurements = [mk_measurement(m) for m in measurements]
+
+    definition = dict(
+        name=metadata_definition['name'],
+        description=metadata_definition['description'],
+        metadata_type='eo',
+        metadata={},
+        measurements=measurements
+    )
+
+    if storage is not None:
+        definition['storage'] = storage
+
+    return definition
+
+
+def _get_product_info(metadata_definition):
+    fields = toolz.get_in(['dataset'], metadata_definition, {})
+    return {name: parse_dataset_field(value, name=name) for name, value in fields.items()
+            if name not in ('search_fields', 'grid_spatial')}
+
+
+def _convert_datetime(val):
+    try:
+        return val.strftime("%Y-%m-%d %H:%M:%S.%f")
+    except AttributeError:
+        return str(val)
+
+
+def _get_boundingbox(metadata_fname):
+    with open(str(metadata_fname)) as fl:
+        metadata = yaml.load(fl)
+    crs = CRS(metadata['crs'])
+    geo = Geometry(metadata['geometry'], crs=crs)
+    spatial_reference = str(crs.wkt)
+
+    left, bottom, right, top = geo.to_crs(CRS('EPSG:4326')).boundingbox
+    lat_lon_coords = {
+        'ul': {'lat': left, 'lon': top},
+        'll': {'lat': left, 'lon': bottom},
+        'ur': {'lat': right, 'lon': top},
+        'lr': {'lat': right, 'lon': bottom},
+        }
+
+    left, bottom, right, top = geo.boundingbox
+    xy_geo_ref_points = {
+        'ul': {'x': left, 'y': top},
+        'll': {'x': left, 'y': bottom},
+        'ur': {'x': right, 'y': top},
+        'lr': {'x': right, 'y': bottom},
+        }
+
+    return xy_geo_ref_points, spatial_reference, lat_lon_coords
+
+
+def _create_new_dataset_def(input_file: Path):
+    """
+    Create new metadata dataset definition yaml files
+    """
+    now = datetime.utcnow()
+    creation_dt = now.strftime('%Y-%m-%dT%H:%M:%S.%f')
+
+    xy_geo_ref_points, spatial_ref, lat_lon_coords = _get_boundingbox(input_file)
+
+    def _get_measurements(m):
+        m_dict = {}
+        for band, img_path in m.items():
+            m_dict[band] = {'path': str(img_path['path']), 'layer': 1}
+        return m_dict
+
+    dataset_def_template = {
+        'id': str(uuid.uuid4()),
+        'creation_dt': creation_dt,
+        'label': '',
```

Comment:
The new-style metadata has a label in the naming conventions, so we could:
```suggestion
        'label': input_file.name.split('.')[0],
```
(or otherwise: it's better to remove it than use the blank string)